src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -918,7 +918,13 @@ private static ulong TransferDataFromRequestStream(IntPtr buffer, int length, Ea
                 else // sts == null
                 {
                     // Allocate a transfer state object to use for the remainder of this request.
-                    easy._sendTransferState = sts = new EasyRequest.SendTransferState();
+                    Debug.Assert(easy._requestMessage.Content != null, "Content shouldn't be null, since we already got a content request stream");
+                    long bufferSize = easy._requestMessage.Content.Headers.ContentLength.GetValueOrDefault();
+                    if (bufferSize <= 0 || bufferSize > RequestBufferSize)
+                    {
+                        bufferSize = RequestBufferSize;
```

Comment:
Would it make sense to rename `RequestBufferSize` to something like `MaxRequestBufferSize`?